src/plugins/view-mode.ts

```diff
@@ -1,7 +1,13 @@
-export default defineNuxtPlugin(() => {
+export default defineNuxtPlugin((nuxtApp) => {
+  const isMobile = ref(false);
+  nuxtApp.hook("app:mounted", () => {
+    watchEffect(() => {
+      isMobile.value = !useMediaQuery("(min-width: 700px)").value;
```

Comment:
Just studying this now, why is this a negated `min-width` query? Would make more sense to me if this was a positive `max-width` query:

```js
isMobile.value = useMediaQuery("(max-width: 699px)").value;
```

And since you encountered unexpected hydration issues, I would add a comment above the `hook` to describe why this `hook` + `watchEffect` is needed.

Reply:
The media query part is a migration of this piece from the Nuxt 2 version:
```
      if (window.matchMedia('(min-width: 700px)').matches) {
        this.$store.commit('setMobileState', false)
      } else {
        this.$store.commit('setMobileState', true)
      }
```
:)
I'm not sure I like a separate literal value for a threshold more than a condition negation.
Currently the project has 23 mentions of 700px and 2 mentions of 699px. I would rather rewrite those 2 to use the same value, otherwise I wouldn't have been smart enough to correctly change if everywhere if I decided to alter the threshold (ok maybe I would at the moment only because we're discussing this). Of course, preferably all that should also move to some constant (CSS custom property, SASS variable, whatever). Separate ticket? :)